### ISSUE TITLE
Only requests is detected as third party

### ISSUE THREAD

**Issue by WhyNotHugo** [timothycrosley/isort#1209]
For some reason, most of my libraries are identified as first-party:

```python
import logging
import math
from decimal import Decimal
from typing import Any, Iterable

import requests

from django.conf import settings
from django.db import models
from django.db.models import Q
from django.utils.functional import cached_property
from django.utils.translation import gettext_lazy as _
from idf.core.models.mixins import UUIDModel
from idf.core.models.payment import PaymentMethod
from ocaclient import OcaClient
from polymorphic.models import PolymorphicModel
from polymorphic.query import PolymorphicQuerySet
```

Here's what I have installed (`pip freeze`):
```
aiohttp==3.6.2
aiohttp-socks==0.3.9
aiorpcX==0.18.4
apipkg==1.5
appdirs==1.4.4
argcomplete==1.11.1
asn1crypto==1.3.0
aspy.refactor-imports==1.1.0
asttokens==2.0.3
async-timeout==3.0.1
atomicwrites==1.4.0
attrs==19.3.0
Automat==20.2.0
awscli==1.18.60
backcall==0.1.0
bcrypt==3.1.7
Beaker==1.11.0
black==19.10b0
botocore==1.16.10
btrfsutil==1.2.0
CacheControl==0.12.6
cached-property==1.5.1
cachy==0.3.0
certifi==2019.11.28
cffi==1.14.0
cfgv==3.1.0
chardet==3.0.4
cleo==0.7.6
click==7.1.2
click-log==0.3.2
click-threading==0.4.4
clikit==0.4.2
colorama==0.4.3
commonmark==0.9.1
configobj==5.0.6
constantly==15.1.0
contextlib2==0.6.0.post1
cryptography==2.9.2
daemonize==2.5.0
decorator==4.4.2
deluge==2.0.4.dev23
distlib==0.3.0
distro==1.5.0
dnspython==1.16.0
docker==4.2.0
docker-compose==1.25.5
docker-pycreds==0.4.0
dockerpty==0.4.1
docopt==0.6.2
docutils==0.16
ecdsa==0.15
Electrum==3.3.8
entrypoints==0.3
filelock==3.0.12
flake8==3.8.1
flake8-bugbear==19.3.0
flake8-comprehensions==3.2.2
flake8-import-order==0.18.1
future==0.18.2
GeoIP==1.3.2
gpg==1.13.1
greenlet==0.4.15
gunicorn==20.0.4
html5lib==1.0.1
humanize==0.5.1
hyperlink==19.0.0
icalendar==4.0.6
identify==1.4.15
idna==2.9
importlib-metadata==1.5.2
incremental==17.5.0
iniconfig==1.0.0
ipython==7.14.0
ipython-genutils==0.1.0
isc==2.0
isort==4.3.21
jedi==0.17.0
jeepney==0.4.3
jmespath==0.10.0
jsonrpclib-pelix==0.4.1
jsonschema==3.2.0
keyring==21.2.0
khal==0.10.1
langdetect==1.0.8
lit==0.10.0.dev0
lockfile==0.12.2
louis==3.13.0
Mako==1.1.2
MarkupSafe==1.1.1
mccabe==0.6.1
msgpack==1.0.0
multidict==4.7.6
mypy==0.770
mypy-extensions==0.4.3
nodeenv==1.3.5
notify2==0.3.1
numpy==1.18.4
oauthlib==3.1.0
openrazer==2.8.0
openrazer-daemon==2.8.0
ordered-set==3.1.1
packaging==20.3
paramiko==2.7.1
parsedatetime==2.5
parso==0.7.0
pastel==0.2.0
pathspec==0.8.0
pbkdf2==1.3
pep517==0.8.2
pew==1.2.0
pexpect==4.8.0
pickleshare==0.7.5
Pillow==6.2.1
pipenv==2018.11.15.dev0
pkginfo==1.5.0.1
pluggy==0.13.1
ply==3.11
poetry==1.0.5
pre-commit==2.2.0
progress==1.5
progressbar2==3.51.3
prompt-toolkit==3.0.5
protobuf==3.11.4
psutil==5.7.0
ptyprocess==0.6.0
pwquality==1.4.2
py==1.8.1
pyaes==1.6.1
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycairo==1.19.1
pycodestyle==2.6.0
pycparser==2.20
pycryptodome==3.9.7
pycryptodomex==3.9.7
pyflakes==2.2.0
Pygments==2.6.1
PyGObject==3.36.1
PyHamcrest==1.9.0
pylev==1.3.0
pymediainfo==4.2.1
PyNaCl==1.3.0
pynvim==0.4.0
pyOpenSSL==19.1.0
pyparsing==2.4.7
PyQt5==5.14.2
PyQt5-sip==12.7.2
pyrsistent==0.16.0
PySocks==1.7.1
python-dateutil==2.8.1
python-libtorrent==1.2.6
python-utils==2.4.0
pytoml==0.1.21
pytz==2020.1
pyudev==0.22.0.dev20200219
pyxdg==0.26
PyYAML==5.3.1
qrcode==6.1
Reflector==2020.3.21.11.40.36
regex==2020.5.7
rencode==1.0.6
reorder-python-imports==2.2.0
requests==2.23.0
requests-oauthlib==1.3.0
requests-toolbelt==0.9.1
retrying==1.3.3
rsa==4.0
s3transfer==0.3.3
schedule==0.6.0
SecretStorage==3.1.2
service-identity==18.1.0
setproctitle==1.1.10
shellingham==1.3.1
sip==4.19.22
six==1.14.0
solaar==0.9.2
speedtest-cli==2.1.2
stevedore==1.32.0
subdownloader==2.1.0
tabulate==0.8.7
team==1.0
texttable==1.6.2
todoman==3.7.0
toml==0.10.1
tomlkit==0.6.0
torbrowser-launcher==0.3.2
tox==3.15.0
traitlets==4.3.3
trash-cli==0.17.1.14
Twisted==20.3.0
txdbus==1.1.0
typed-ast==1.4.1
typing-extensions==3.7.4.1
tzlocal==2.1
urllib3==1.25.9
urwid==2.1.0
vdirsyncer==0.16.7
virtualenv==20.0.18
virtualenv-clone==0.5.4
virtualenvwrapper==4.8.4
wcwidth==0.1.9
webencodings==0.5.1
websocket-client==0.57.0
yarl==1.4.2
youtube-dl==2020.5.8
zipp==3.1.0
zope.interface==5.1.0

```

**Comment by mfogel** [comment]
I'm not an isort dev, but I had this problem as well. It seems that by default isort considers libraries 'firstparty' when it can't otherwise tell. I fixed this in my project by setting

```
[tool.isort]
known_first_party = "mylib1,mylib2"
default_section = "THIRDPARTY"
```

in my `pyproject.toml`.

**Comment by at3560k** [comment]
I just tripped over what I think was a "bug" near identical to this - and glad for the above post.

I confirm setting known_first_part and default_section fixed this (and left imports compatible with 'black' style) for me

Here's some more information about my stranger environment that might assist in reproducing the cause (a bit gross)

- pyenv -> 
    -> "local tools environment" (3.8 with tox-pyenv, and isort, plugins)
    -> "python3.8" as py38
    -> (others)

and what I observed was that:

- invoking isort *locally* "isort --verbose --check-only  --recursive src/" passed
- invoking isort via abspath (like tox did) "/home/myuser/REPO/project/project/.tox/check/bin/isort --verbose --check-only ... " *failed* by misdetecting software installed in the nested virtual environment as firstparty and demanded a change.

Here's a snippet of the log without "default_section = thirdparty" added (known_first_party did not seem to make the difference).  Please forgive the bawderlization of my username and the strange looking repository structure :). 


```bash

$ isort --verbose --check-only --diff --skip tests --recursive src setup.py | less
...

SUCCESS: /home/USER/REPO/my-lib/my-lib/src/my_lib/__version__.py Everything Looks Good!
from-type place_module for flask returned THIRDPARTY
from-type place_module for flask_restx returned THIRDPARTY
from-type place_module for my_lib.apis.controllers.namespace_artifacts returned FIRSTPARTY
from-type place_module for my_lib.apis.controllers.namespace_builds returned FIRSTPARTY
from-type place_module for my_lib.apis.controllers.namespace_components returned FIRSTPARTY
SUCCESS: /home/USER/REPO/my-lib/my-lib/src/my_lib/apiv1.py Everything Looks Good!
...

 $ .tox/check/bin/isort --verbose --check-only --diff --skip tests --recursive src setup.py | less
 
 SUCCESS: /home/USER/REPO/my-lib/my-lib/src/my_lib/__version__.py Everything Looks Good!
 from-type place_module for flask returned FIRSTPARTY
 from-type place_module for flask_restx returned FIRSTPARTY
 from-type place_module for my_lib.apis.controllers.namespace_artifacts returned FIRSTPARTY
 from-type place_module for my_lib.apis.controllers.namespace_builds returned FIRSTPARTY
 from-type place_module for my_lib.apis.controllers.namespace_components returned FIRSTPARTY
 ERROR: /home/USER/REPO/my-lib/my-lib/src/my_lib/apiv1.py Imports are incorrectly sorted.
 --- /home/USER/REPO/my-lib/my-lib/src/my_lib/apiv1.py:before       2020-06-10 13:17:32.320190
 +++ /home/USER/REPO/my-lib/my-lib/src/my_lib/apiv1.py:after        2020-06-19 13:51:13.821393
 @@ -10,16 +10,15 @@
  # Core modules


 -# Installed modules
 -# ---------------------------------------------------------
 -from flask import Blueprint
 -from flask_restx import Api
 -
  # local modules
  # ---------------------------------------------------------
  from my_lib.apis.controllers.namespace_artifacts import api as ns_art1
```

I suspect at least in my case, there's some interaction going on between tox/pyenv.  But in both virtualenvs the .egg-link is identical.

**Comment by timothycrosley** [comment]
These type of issues should be fixed in the latest 5.0.0 release of isort: https://timothycrosley.github.io/isort/docs/major_releases/introducing_isort_5/

Thanks for everyone for reporting them to help me prioritize!

~Timothy
